app/javascript/mastodon/components/status_banner.tsx

```diff
@@ -10,33 +13,52 @@ export const StatusBanner: React.FC<{
   variant: BannerVariant;
   expanded?: boolean;
   onClick?: () => void;
-}> = ({ children, variant, expanded, onClick }) => (
-  <label
-    className={
-      variant === BannerVariant.Warning
-        ? 'content-warning'
-        : 'content-warning content-warning--filter'
+}> = ({ children, variant, expanded, onClick }) => {
+  const descriptionId = useId();
+
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  const forwardClick = useCallback<MouseEventHandler>((e) => {
+    if (buttonRef.current && e.target !== buttonRef.current) {
+      buttonRef.current.click();
     }
-  >
-    {children}
+  }, []);
+
+  return (
+    // Element clicks are passed on to button
+    // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
+    <div
+      className={
+        variant === BannerVariant.Warning
+          ? 'content-warning'
+          : 'content-warning content-warning--filter'
+      }
+      onClick={forwardClick}
+    >
+      <p id={descriptionId}>{children}</p>
 
-    <button className='link-button' onClick={onClick}>
-      {expanded ? (
-        <FormattedMessage
-          id='content_warning.hide'
-          defaultMessage='Hide post'
-        />
-      ) : variant === BannerVariant.Warning ? (
-        <FormattedMessage
-          id='content_warning.show_more'
-          defaultMessage='Show more'
-        />
-      ) : (
-        <FormattedMessage
-          id='content_warning.show'
-          defaultMessage='Show anyway'
-        />
-      )}
-    </button>
-  </label>
-);
+      <button
+        ref={buttonRef}
+        className='link-button'
+        onClick={onClick}
+        aria-describedby={descriptionId}
```

Comment:
> The `aria-describedby` property is appropriate when the associated content contains plain text. If the content is extensive, contains useful semantics, or has a complex structure requiring user navigation, use [aria-details](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-details) instead.

Content warnings are mostly textual but they are allowed to include emojis, including custom ones, so I wonder if we should use `aria-details` instead?

Reply:
The name (shortcode) of custom emojis is read out normally, so I think `aria-describedby` works here:

https://github.com/user-attachments/assets/775f9180-c666-48bb-abec-711f7b479807

In the future it would be good to allow defining dedicated alt text for custom emojis where the shortcode might not be very descriptive.